TCC.Lib/Benchmark/BenchmarkHelper.cs

```diff
@@ -0,0 +1,129 @@
+                        {
+                            Algo = algo,
+                            CompressionRatio = ratio,
+                            Encryption = encrypt,
+                        };
+                    }
+                }
+            }
+        }
+
+        public static async Task<OperationSummary> RunBenchmark(BenchmarkOption benchmarkOption, CancellationToken cts)
+        {
+
+            //var inputFolder = TestFileHelper.NewFolder();
+
+            var keysFolder = TestFileHelper.NewFolder();
+
+            //var testFile = TestFileHelper.NewFile(inputFolder, fileSizeMb);
+            FileInfo src = new FileInfo(benchmarkOption.Source);
+
+            var iterations = benchmarkOption.GenerateBenchmarkIteration().ToList();
+
+            foreach (var iter in iterations)
+            {
+                PasswordMode pm = iter.Encryption ? PasswordMode.PublicKey : PasswordMode.None;
+                var compressedFolder = TestFileHelper.NewFolder();
+                var outputFolder = TestFileHelper.NewFolder();
+
+                // compress
+                var compressOption = new CompressOption
+                {
+                    Algo = iter.Algo,
+                    CompressionRatio = iter.CompressionRatio,
+                    BlockMode = BlockMode.Explicit,
+                    SourceDirOrFile = benchmarkOption.Source,
+                    DestinationDir = compressedFolder,
+                    Threads = Environment.ProcessorCount,
+                    PasswordOption = await BenchmarkOptionHelper.GenerateCompressPassswordOption(pm, keysFolder)
+                };
+
+                Stopwatch swComp = Stopwatch.StartNew();
+
+                var resultCompress = await TarCompressCrypt.Compress(compressOption, cancellationToken: cts);
+
+                swComp.Stop();
+                if (!resultCompress.IsSuccess)
+                {
+                    throw new Exception();
+                }
+
+                var compressedFile = resultCompress.Blocks.Select(i => i.DestinationArchive).FirstOrDefault();
+                FileInfo fi = new FileInfo(compressedFile);
+
+                double compressionFactor = (double)src.Length / (double)fi.Length;
+
+                // decompress
+                var decompressOption = new DecompressOption
+                {
+                    SourceDirOrFile = compressedFile,
+                    DestinationDir = outputFolder,
+                    Threads = Environment.ProcessorCount,
+                    PasswordOption = BenchmarkOptionHelper.GenerateDecompressPasswordOption(pm, keysFolder)
+                };
+
+                Stopwatch swDecomp = Stopwatch.StartNew();
+                var resultDecompress = await TarCompressCrypt.Decompress(decompressOption, cancellationToken: cts);
+                swDecomp.Stop();
+                if (!resultDecompress.IsSuccess)
+                {
+                    throw new Exception();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') 'System.Exception' should not be thrown by user code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS112)